tests/Application.Tests/ExecutorTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace Symbolica.Application
+{
+    public class ExecutorTests
+    {
+        [Theory]
+        [ClassData(typeof(FailTestData))]
+        private async Task ShouldFail(string directory,
+            bool useSymbolicGarbage, bool useSymbolicAddresses, bool useSymbolicContinuations)
+        {
+            var result = await Executor.Run(directory,
+                useSymbolicGarbage, useSymbolicAddresses, useSymbolicContinuations);
+
+            result.IsSuccess.Should().BeFalse();
+            result.Exception.Space.GetExample().Select(p => p.Key).Should().BeEquivalentTo(
+                await File.ReadAllLinesAsync(Path.Combine(directory, "symbols")));
+        }
+
+        [Theory]
+        [ClassData(typeof(PassTestData))]
+        private async Task ShouldPass(string directory,
+            bool useSymbolicGarbage, bool useSymbolicAddresses, bool useSymbolicContinuations)
+        {
+            var result = await Executor.Run(directory,
+                useSymbolicGarbage, useSymbolicAddresses, useSymbolicContinuations);
+
+            result.IsSuccess.Should().BeTrue();
+        }
+
+        private sealed class FailTestData : TestData
+        {
+            public FailTestData()
+                : base("fail")
+            {
+            }
+        }
+
+        private sealed class PassTestData : TestData
+        {
+            public PassTestData()
+                : base("pass")
+            {
+            }
+        }
+
+        private class TestData : TheoryData<string, bool, bool, bool>
+        {
+            protected TestData(string status)
+            {
+                foreach (var directory in Directory.EnumerateDirectories(Path.Combine("..", "..", "..", "..", status)))
+                {
+                    Add(directory, false, false, false);
```

Comment:
cartesian product with sql style linq?

Reply:
I just chucked 3 `foreach` loops in there.

![image](https://user-images.githubusercontent.com/19819531/126668113-4b778091-5b27-4708-bc5b-6d6cee996813.png)